forum/notifications/api_view.py

```diff
@@ -0,0 +1,272 @@
+from rest_framework import viewsets, status
+from rest_framework.response import Response
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.decorators import action
+from rest_framework.views import APIView
+from .models import Notification, StartupNotificationPreferences, InvestorNotificationPreferences, Startup, Investor
+from .serializers import NotificationSerializer, StartupNotificationPrefsSerializer, InvestorNotificationPrefsSerializer, TriggerNotificationSerializer
+from .tasks import trigger_notification_task
+from .permissions import IsInvestor, IsStartup
+from django.core.exceptions import ValidationError
+
+
+def create_error_response(message, status_code):
+    """
+    Creates a standardized error response.
+
+    Args:
+        message (str or dict): Error message or dictionary of errors.
+        status_code (int): HTTP status code for the response.
+
+    Returns:
+        Response: A DRF Response object with the error message and status code.
+    """
+    if isinstance(message, str):
+        return Response({'error': message}, status=status_code)
+    return Response(message, status=status_code)
+
+
+def get_user_role_and_object(user):
+    """
+    Retrieves the roles and corresponding objects (Investor, Startup) for the given user.
+
+    Args:
+        user: The authenticated user.
+
+    Returns:
+        dict: A dictionary with roles ('investor', 'startup') as keys and corresponding objects as values.
+    """
+    roles_and_objects = {}
+
+    if user.roles.filter(name='investor').exists():
+        try:
+            investor = Investor.objects.get(user=user)
+            roles_and_objects['investor'] = investor
+        except Investor.DoesNotExist:
+            roles_and_objects['investor'] = None
+
+    if user.roles.filter(name='startup').exists():
+        try:
+            startup = Startup.objects.get(user=user)
+            roles_and_objects['startup'] = startup
+        except Startup.DoesNotExist:
+            roles_and_objects['startup'] = None
+
+    return roles_and_objects if roles_and_objects else None
```

Comment:
It seems that `roles_and_objects` is sometimes returning `None`. This could result in issues down the line if other parts of your code expect a dictionary. Return an empty dictionary rather than `None` when no roles are found to avoid unexpected behavior.